module/VuFind/src/VuFind/View/Helper/Root/EscapeOrCleanHtml.php

```diff
@@ -71,15 +71,16 @@ public function __construct(protected Escaper $escaper, protected CleanHtml $cle
      * @param ?bool   $allowHtml        Whether to allow sanitized HTML if passed a PropertyString
      * @param string  $renderingContext Rendering context for cleaning HTML
      *
-     * @return mixed Given a string, returns an escaped string, otherwise returns self
+     * @return mixed Called with non-null arguments, returns an escaped string, otherwise returns self
      */
     public function __invoke(
         $value = null,
         ?string $dataContext = null,
         ?bool $allowHtml = null,
         string $renderingContext = 'default'
     ) {
-        if (null === $value) {
+        // Return self if all arguments are default:
+        if (null === $value && null === $dataContext && null === $allowHtml && 'default' === $renderingContext) {
```

Comment:
It's hard to judge this without understanding / remembering why the invoke method ever returned $this for null values in the first place.  I see the [commit](https://github.com/vufind-org/vufind/pull/4004/commits/efa191a02d749dcabc4213202c765d7b63eddf4f#diff-c857983c15e81b73101a774801c855a7eb0adfdddc4616f0b3442d248310ba1c) where that was added but not why it was done.  So I think it probably makes sense? but not sure.

Reply:
It's a common convention in Laminas view helpers, but if we don't need it, we should definitely consider removing it.

Reply:
...and now the PR has been revised to remove it.